OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs

```diff
@@ -231,11 +231,13 @@ public SaveDirectory(Folder folder, string displayName, MapClassification classi
 			};
 
 			var save = widget.Get<ButtonWidget>("SAVE_BUTTON");
+			save.IsDisabled = () => string.IsNullOrWhiteSpace(filename.Text) || string.IsNullOrWhiteSpace(title.Text) || string.IsNullOrWhiteSpace(author.Text);
+
+			if (string.IsNullOrEmpty(filename.Text))
+				filename.TakeKeyboardFocus();
```

Comment:
Could you please move this closer to where `filename.Text` is populated (right after line 162)?